generator/routing_index_generator.cpp

```diff
@@ -68,28 +69,35 @@ class VehicleMaskBuilder final
 
   VehicleMask CalcRoadMask(FeatureType & f) const
   {
-    return CalcMask(f, [&](VehicleModelInterface const & model, FeatureType & f) {
-      return model.IsRoad(f);
+    feature::TypesHolder const types(f);
+    if (types.HasWithSubclass(m_constructionType))
+      return 0;
+
+    return CalcMask([&](VehicleModelInterface const & model)
+    {
+      return model.IsRoad(types);
     });
   }
 
   VehicleMask CalcOneWayMask(FeatureType & f) const
   {
-    return CalcMask(f, [&](VehicleModelInterface const & model, FeatureType & f) {
-      return model.IsOneWay(f);
+    feature::TypesHolder const types(f);
+    return CalcMask([&](VehicleModelInterface const & model)
+    {
+      return model.IsOneWay(types);
     });
```

Comment:
The current syntax looks strange, maybe use this one?
```suggestion
        {
          return model.IsOneWay(types);
        });
```

Reply:
Well, I used to write lambdas like this and I think it is comfortable. When lambda's function is +1 offset from main code. No need to make any _strange_ alignment and put 3-4 offsets:
```
prologue;
foo([&]()
{
  // lambda code with 1 offset
});
epilogue;
```
vs
```
prologue;
foo([&]()
       {
         // lambda code with 5 offsets
       });
epilogue;

```